File: ai-path/02-ai-agent-engineer/final-project/scripts/check.py
"""验证 ai-agent-platform 项目结构。"""
import sys
from pathlib import Path

REQUIRED_FILES = [
    "backend/app/main.py",
    "backend/app/agent/engine.py",
    "backend/app/rag/pipeline.py",
    "backend/app/workflow/engine.py",
    "backend/app/api/v1/chat.py",
    "backend/app/api/v1/agents.py",
    "backend/app/api/v1/knowledge.py",
    "backend/app/api/v1/workflows.py",
    "backend/app/api/v1/skills.py",
    "backend/app/api/v1/auth.py",
    "backend/app/core/config.py",
    "backend/app/core/database.py",
    "backend/app/core/security.py",
    "backend/requirements.txt",
    "backend/Dockerfile",
    "frontend/package.json",
    "frontend/src/main.ts",
    "frontend/src/App.vue",
    "frontend/src/views/Chat.vue",
    "frontend/src/views/Agents.vue",
    "frontend/src/views/Knowledge.vue",
    "frontend/src/views/Workflows.vue",
    "frontend/src/views/Skills.vue",
    "docker-compose.yml",
    ".env.example",
    "README.md",
]

def main():
    root = Path(__file__).resolve().parent.parent / "ai-agent-platform"
    errors = []
    for rel in REQUIRED_FILES:
        if not (root / rel).exists():
            errors.append(f"缺失: {rel}")

    if errors:
        for e in errors:
            print(f"  ✗ {e}")
        sys.exit(1)
    print("  ✓ ai-agent-platform 结构验证通过")
    sys.exit(0)

if __name__ == "__main__":
    main()

File: backend/nodejs-backend/final-project/scripts/check.js
/**
 * 验证 blog-api 项目的目录结构和必需文件。
 *
 * 用法: node scripts/check.js
 */
const fs = require("fs");
const path = require("path");

const root = path.resolve(__dirname, "..");

const REQUIRED_FILES = [
  "src/app.ts",
  "src/config/index.ts",
  "src/routes/index.ts",
  "src/modules/auth/auth.routes.ts",
  "src/modules/auth/auth.service.ts",
  "src/modules/auth/auth.controller.ts",
  "src/modules/auth/auth.schema.ts",
  "src/modules/user/user.routes.ts",
  "src/modules/user/user.service.ts",
  "src/modules/user/user.controller.ts",
  "src/modules/post/post.routes.ts",
  "src/modules/post/post.service.ts",
  "src/modules/post/post.controller.ts",
  "src/modules/post/post.schema.ts",
  "src/modules/comment/comment.routes.ts",
  "src/modules/comment/comment.service.ts",
  "src/modules/comment/comment.controller.ts",
  "src/modules/comment/comment.schema.ts",
  "src/modules/category/category.routes.ts",
  "src/modules/category/category.service.ts",
  "src/modules/category/category.controller.ts",
  "src/modules/category/category.schema.ts",
  "src/modules/tag/tag.routes.ts",
  "src/modules/tag/tag.service.ts",
  "src/modules/tag/tag.controller.ts",
  "src/modules/tag/tag.schema.ts",
  "src/middleware/auth.ts",
  "src/middleware/role.ts",
  "src/middleware/validate.ts",
  "src/middleware/errorHandler.ts",
  "src/middleware/rateLimiter.ts",
  "src/middleware/upload.ts",
  "src/lib/prisma.ts",
  "src/lib/redis.ts",
  "src/lib/logger.ts",
  "src/utils/slug.ts",
  "src/utils/response.ts",
  "src/utils/errors.ts",
  "prisma/schema.prisma",
  "docker-compose.yml",
  "Dockerfile",
  "tsconfig.json",
  "package.json",
  ".env.example",
];

const FORBIDDEN_FILES = [".env", "logs/combined.log", "logs/error.log"];

let errors = [];
let warnings = [];

// 检查必需文件
for (const rel of REQUIRED_FILES) {
  if (!fs.existsSync(path.join(root, rel))) {
    errors.push(`缺失: ${rel}`);
  }
}

// 检查不应提交的文件
for (const rel of FORBIDDEN_FILES) {
  if (fs.existsSync(path.join(root, rel))) {
    warnings.push(`警告: ${rel} 不应提交到仓库（已在 .gitignore 中）`);
  }
}

// 检查 Prisma schema 中的模型
const schemaPath = path.join(root, "prisma", "schema.prisma");
if (fs.existsSync(schemaPath)) {
  const schema = fs.readFileSync(schemaPath, "utf-8");
  const requiredModels = ["User", "Post", "Comment", "Category", "Tag"];
  for (const model of requiredModels) {
    if (!schema.includes(`model ${model}`)) {
      errors.push(`Prisma schema 缺少模型: ${model}`);
    }
  }
}

// 输出结果
if (errors.length > 0) {
  errors.forEach((e) => console.log(`  ✗ ${e}`));
}
if (warnings.length > 0) {
  warnings.forEach((w) => console.log(`  ⚠ ${w}`));
}
if (errors.length === 0) {
  console.log("  ✓ blog-api 结构验证通过");
  process.exit(0);
} else {
  process.exit(1);
}
